app/models/markdown_archive.rb

```diff
@@ -49,6 +50,21 @@ def add_article(tmp, article)
     File.open(tmp.join(path), "w"){ |f| f.write(doc) }
   end
 
+  def add_topic(tmp, topic)
+    doc = TopicsController.render(
+      template: "topics/show.md.erb",
+      assigns: { topic: topic }
+    )
+
+    path = [
+      "Topics",
```

Comment:
Since the intro and lessons are all tabs on the same page, I think in the export it might make more sense if they were all found in the same directory. Maybe it'd be better to group files by topic too, rather than dump everything in Lessons like I did? So something more like this:

```
Topics/Passwords/Intro.md
Topics/Passwords/Beginning.md
Topics/Passwords/Intermediate.md
...
```

Reply:
Thank you! I updated with that suggestion and the ones from our conversation about the test file.